### DISCUSSION TITLE
"play all" button

### DISCUSSION THREAD

**Discussion by gitbreaker222** [feugy/melodie#23] [Ideas]
as a user I want to put my complete library into the queue. I can use my music pool as my personal favourite radio station when randomly play through everything I've collected

**Comment by feugy** [comment]
Hello @gitbreaker222, and thanks for your interest in Mélodie!

I like your idea: it's raising an interesting challenge. Enqueued tracks are visually represented with draggable, animatable components. When you shuffle your tracks queue, or when you clear it, you can see them moving around.
Unfortunately, these components are a bit heavy, and the application lags when there are too of them. It depends on your laptop, but it's pretty obvious on mine above a hundred tracks.

My library is around 10 000 tracks, and enqueuing them would kill electron. Out of curiosity, how big is your library?
I see several ways to implement your idea:
- try to optimize the components to make them lighter (conditional animations...)
- introduce virtualization to only display "visible" tracks in the queue
- load (and unload) tracks in batches, to keep a "reasonable" number of them alive

**Reply by gitbreaker222** [reply]
thanks for your reply :) 

My library has 8936 songs so I'd expect the same performance issues, if they are all loaded to the DOM. The best solution I know is to use a virtual list. This of course means, that animating list entries outside the viewport would be very tricky.

Sometimes I work on my playlist-mechanics demo and when I switched to a virtual list, I removed the animations in favour for supporting large lists. In svelte it possible to give items a default animation, if they don't transition between two places in the DOM (e.g. translate 100px down + fade out). I hope to update that soon here:

https://svelte.dev/repl/73117a195fa2433395d573859893c04a?version=3.32.3

Maybe it is also an option, to replace the shuffle animation with real items with a "presentational" animation covering the list.  For my dice rolling this is sufficient I guess ^^ https://codepen.io/Breaker222/full/Jxarpz

Also I just realize, this is a svelte project: I happen to be a little contributor for the [svelte-virtual-list](https://github.com/sveltejs/svelte-virtual-list/pull/52), so if the items can have the same height (no more line brakes for overflowing titles), my proposed edits work fine :)

**Comment by feugy** [comment]
Hi Gitbreaker222.

Thanks for the snippets, they'll be very handy.
I'm currently in the middle of an heavy refactoring, as I'm implementing the ability to run Mélodie on the web, and to broadcast medias.

I'm getting some performance issues on this new version, and I'm closely looking at list virtualization for the albums & artists views, in addition to the track list itself.

Be sure I'm not forgetting your request.
